builtin/providers/aws/resource_aws_security_group_test.go

```diff
@@ -53,12 +72,21 @@ func TestResourceAwsSecurityGroupIPPermGather(t *testing.T) {
 			"from_port": int64(80),
 			"to_port":   int64(80),
 			"security_groups": schema.NewSet(schema.HashString, []interface{}{
-				"foo",
+				"sg-22222",
+			}),
+		},
+		map[string]interface{}{
+			"protocol":  "tcp",
+			"from_port": int64(443),
+			"to_port":   int64(443),
+			"security_groups": schema.NewSet(schema.HashString, []interface{}{
+				"ec2_classic",
+				"amazon-elb/amazon-elb-sg",
```

Comment:
This diff is strange, it doesn't highlight the changes in the `raw` struct above, starting on line 9. It's clearly different from [master][1]. I think it makes more sense if you see the entire diff

This chuck matches up with the newly added `ec2.IpPermission`  block:

```
		&ec2.IpPermission{
			IpProtocol: aws.String("tcp"),
			FromPort:   aws.Int64(int64(443)),
			ToPort:     aws.Int64(int64(443)),
			UserIdGroupPairs: []*ec2.UserIdGroupPair{
				// Classic
				&ec2.UserIdGroupPair{
					UserId:    aws.String("12345"),
					GroupId:   aws.String("sg-33333"),
					GroupName: aws.String("ec2_classic"),
				},
				&ec2.UserIdGroupPair{
					UserId:    aws.String("amazon-elb"),
					GroupId:   aws.String("sg-d2c979d3"),
					GroupName: aws.String("amazon-elb-sg"),
				},
			},
		}
```

It represents a security group with two nam references, one to a group named `ec2_classic` and another is the special Amazon ELB security group. 



1: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_security_group_test.go#L18